qa/tasks/cephfs/mount.py

```diff
@@ -233,7 +233,7 @@ def _cleanup_netns(self):
                 "ip link set brx.{0} down".format(self.nsid)]
         self.client_remote.run(args=args, timeout=(5*60), omit_sudo=False)
         args = ["sudo", "bash", "-c",
-                "ip link delete brx.{0}".format(self.nsid)]
+                "ip link delete dev brx.{0} master ceph-brx".format(self.nsid)]
```

Comment:
My manpage says that the ip link delete command has a format like this:

    ip link delete { DEVICE | group GROUP } type TYPE [ ARGS ]

"master" seems to be a keyword only for `ip link set`. Shouldn't that be instead:

    ip link delete dev brx.{0} type veth

?

Reply:
I'd also add on to this that it's not clear to me how this fixes the problem in the ticket.

Reply:
(Please update the commit message.)

Reply:
> My manpage says that the ip link delete command has a format like this:
> 
> ```
> ip link delete { DEVICE | group GROUP } type TYPE [ ARGS ]
> ```
> 
> "master" seems to be a keyword only for `ip link set`. Shouldn't that be instead:
> 
> ```
> ip link delete dev brx.{0} type veth
> ```
> 
> ?

Yeah, will fix it.

Reply:
> I'd also add on to this that it's not clear to me how this fixes the problem in the ticket.

Sorry I can reproduce it sometimes again, after deleting one netns, the network seems will be shortly affected and unreachable, mostly it will last few seconds, but in the tracker case it will last 40+ seconds, then we lost the up:reconnect mdsmap in client:

```
261 2020-06-30T01:00:43.282+0000 7f9ec97fa700 20 client.10650 trim_cache size 0 max 16384
262 2020-06-30T01:00:44.282+0000 7f9ec97fa700 20 client.10650 trim_cache size 0 max 16384
263 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3301/0,v1:172.21.15.146:6790/0] conn(0x7    f9ea8007690 msgr2=0x7f9ea8009b50 unknown :-1 s=STATE_CONNECTING_RE l=1).mark_down
264 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3301/0,v1:172.21.15.146:6790/0] conn(0    x7f9ea8007690 0x7f9ea8009b50 unknown :-1 s=START_CONNECT pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).stop
265 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3300/0,v1:172.21.15.146:6789/0] conn(0x7    f9ea8004a70 msgr2=0x7f9ea8006f10 unknown :-1 s=STATE_CONNECTING_RE l=1).mark_down
266 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3300/0,v1:172.21.15.146:6789/0] conn(0    x7f9ea8004a70 0x7f9ea8006f10 unknown :-1 s=START_CONNECT pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).stop
267 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 >> [v2:172.21.15.63:3300/0,v1:172.21.15.63:6789/0] conn(0x7f9    ea8001f00 msgr2=0x7f9ea8004350 unknown :-1 s=STATE_CONNECTING_RE l=1).mark_down
268 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.63:3300/0,v1:172.21.15.63:6789/0] conn(0x7    f9ea8001f00 0x7f9ea8004350 unknown :-1 s=START_CONNECT pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).stop
269 2020-06-30T01:00:44.781+0000 7f9ed1c59700  1 -- 192.168.0.2:0/2664284312 reap_dead start
270 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3300/0,v1:172.21.15.146:6789/0] conn(0    x7f9ea800ae90 0x7f9ea800d2e0 unknown :-1 s=NONE pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).connect
271 2020-06-30T01:00:44.781+0000 7f9ed1c59700  1 -- 192.168.0.2:0/2664284312 reap_dead start
272 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.63:3300/0,v1:172.21.15.63:6789/0] conn(0x7    f9ea8004f00 0x7f9ea8007b40 unknown :-1 s=NONE pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).connect
273 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 --2- 192.168.0.2:0/2664284312 >> [v2:172.21.15.146:3301/0,v1:172.21.15.146:6790/0] conn(0    x7f9ea8008030 0x7f9ea80042f0 unknown :-1 s=NONE pgs=0 cs=0 l=1 rev1=0 rx=0 tx=0).connect
274 2020-06-30T01:00:44.781+0000 7f9ed1c59700  1 -- 192.168.0.2:0/2664284312 reap_dead start
275 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 --> [v2:172.21.15.63:3300/0,v1:172.21.15.63:6789/0] -- mon_ge    tmap magic: 0 v1 -- 0x7f9ea800a350 con 0x7f9ea8004f00
276 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 --> [v2:172.21.15.146:3300/0,v1:172.21.15.146:6789/0] -- mon_    getmap magic: 0 v1 -- 0x7f9ea800a4c0 con 0x7f9ea800ae90
277 2020-06-30T01:00:44.781+0000 7f9ec1ffb700  1 -- 192.168.0.2:0/2664284312 --> [v2:172.21.15.146:3301/0,v1:172.21.15.146:6790/0] -- mon_    getmap magic: 0 v1 -- 0x7f9ea800a680 con 0x7f9ea8008030
278 2020-06-30T01:00:45.282+0000 7f9ec97fa700 20 client.10650 trim_cache size 0 max 16384
279 2020-06-30T01:00:46.282+0000 7f9ec97fa700 20 client.10650 trim_cache size 0 max 16384
```


Locally I have tried to skip deleting the netnses when doing the umount, and I have tried 300+ times, it works well till now. I have checked the manpage and docs about netns/bridge, still cannot figure out why deleting one netns could affect the others, except the following from netns manual page:

```
              It is possible to lose the physical device when it was moved to netns and then this netns was deleted with a running process:

                 $ ip netns add net0
                 $ ip link set dev eth0 netns net0
                 $ ip netns exec net0 SOME_PROCESS_IN_BACKGROUND
                 $ ip netns del net0

              and eth0 will appear in the default netns only after SOME_PROCESS_IN_BACKGROUND will exit or will be killed. To prevent this the processes
              running in net0 should be killed before deleting the netns:

                 $ ip netns pids net0 | xargs kill
                 $ ip netns del net0
```
I have checked this, in our case I didn't see any existing processes after ceph-fuse unmount in the container.

For the workaround we can base on https://github.com/ceph/ceph/pull/35283 to delay deleting the netnses until the test script exits. And also the netnses maybe created/deleted many times when running the test cases, this will also could speed up the tests.

Reply:
> Locally I have tried to skip deleting the netnses and peer net devices when doing the umount, and I have tried 300+ times, it works well till now. 

Let's just defer deletion of the netns until after the last mount is unmounted. Leave a comment explaining the issue.

Let's  just do the simple solution and move on.

Reply:
Updated it, the first 4 commits are the same with and based on #35283.  The last commit here will defer cleaning the netnses and bridge after the last mountpoint is umounted.